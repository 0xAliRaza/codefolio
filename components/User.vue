<template>
  <div class="user">
    <img
      class="user__image"
      :src="user.image"
      :alt="`${user.fullname}'s image`"
    />
    <h1>{{ user.fullname }}</h1>
    <div>
      {{ user.about }}
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import UserModel from '@/models/User'
export default Vue.extend({
  props: {
    user: {
      required: true,
      type: Object as () => UserModel,
    },
  },
})
</script>
<style lang="scss" scoped>
@import '@/scss/abstracts';
.user {
  display: flex;
  flex-direction: column;
  &__image {
    max-width: pxToRem(200);
    display: block;
    height: auto;
  }
}
</style>
