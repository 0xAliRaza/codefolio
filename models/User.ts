export default interface UserModel {
  fullname: string
  about: string
  image: string
}
