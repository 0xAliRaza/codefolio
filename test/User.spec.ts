import { shallowMount } from '@vue/test-utils'
import User from '@/components/User.vue'

const dummyUser: any = {
  fullname: 'John Doe',
  about:
    "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
  image: 'source.unsplash.com/random/500x500',
}
const factory = () => {
  return shallowMount(User, { propsData: { user: dummyUser } })
}

describe('User', () => {
  test('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = factory()
    expect(wrapper).toMatchSnapshot()
  })

  test('it renders user fullname prop', () => {
    const wrapper = factory()
    expect(wrapper.text()).toContain(dummyUser.fullname)
  })

  test('it renders user about prop', () => {
    const wrapper = factory()
    expect(wrapper.text()).toContain(dummyUser.about)
  })

  test('it renders user image tag', () => {
    const wrapper = factory()
    const img = wrapper.find('img')
    expect(img).toBeTruthy()
    expect(img.attributes('src')).toMatch(dummyUser.image)
  })
})
