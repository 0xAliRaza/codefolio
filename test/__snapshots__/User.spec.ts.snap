// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
